src/test/scala/com/eclipsesource/schema/SchemaValidatorSpec.scala

```diff
@@ -165,39 +167,37 @@ class SchemaValidatorSpec extends PlaySpecification {
       }
     }
 
-  "should resolve references on the classpath via UrlResolver" in {
-    val validator = SchemaValidator().addUrlResolver(ClasspathUrlResolver())
+  "should resolve references on the classpath with ClasspathUrlResolver" in {
+    val validator = SchemaValidator().addUrlResolver(ClasspathUrlProtocolHandler())
     // some.json references location.json within JAR
     val someJson = getClass.getResourceAsStream("/some.json")
     val schema = JsonSource.schemaFromStream(someJson)
     val result = validator.validate(schema.get, Json.obj("location" -> Json.obj("name" -> "Munich")))
     result.isSuccess must beTrue
   }
 
-  "should resolve references on the classpath via UrlResolver and custom protocols (#65)" in {
+  "should resolve relative references on the classpath with relative URL handler (#65)" in {
     val validator = SchemaValidator()
-      .addUrlResolver(ClasspathUrlResolver())
-      .shouldResolveRelativeRefsWithCustomProtocols(true)
-    // some.json references location.json within JAR
+      .addRelativeUrlHandler(new ClasspathHandler)
+    // some-issue-65.json references location.json within JAR
     val someJson = getClass.getResourceAsStream("/some-issue-65.json")
     val schema = JsonSource.schemaFromStream(someJson)
     val result = validator.validate(schema.get, Json.obj("location" -> Json.obj("name" -> "Munich")))
     result.isSuccess must beTrue
   }
 
-  "should resolve relative references with custom protocols (#65)" in {
+  "should resolve relative references on the classpath with relative URL handler (#65 - 2)" in {
     val validator = SchemaValidator()
-      .addUrlResolver(ClasspathUrlResolver())
-      .shouldResolveRelativeRefsWithCustomProtocols(true)
+      .addRelativeUrlHandler(new ClasspathHandler)
     val mySchemaJson = getClass.getResourceAsStream("/my-schema.schema")
     val mySchema = JsonSource.schemaFromStream(mySchemaJson)
     val result = validator.validate(mySchema.get, Json.obj("foo" -> Json.obj("bar" -> "whatever")))
     result.isSuccess must beTrue
   }
 
-  "should fail resolving relative references with custom protocols if option not set(#65)" in {
+  "should fail resolving relative references without any relative URL handler available (#65)" in {
     val validator = SchemaValidator()
-      .addUrlResolver(ClasspathUrlResolver())
+      .addUrlResolver(ClasspathUrlProtocolHandler())
```

Comment:
Thanks for the detailed explanation and examples. It wasn't clear to me that you were referring to the resolution scope/directory of the base schema in the case of relative refs. Unfortunately, the validator currently isn't able to derive the location from which the base schema has been loaded in order to set an initial resolution scope (so you would have to provide it with an explicit "id" property), but this is something that should be taken care of anyway. I'll open issues accordingly.